irohad/synchronizer/impl/synchronizer_impl.cpp

```diff
@@ -138,7 +140,8 @@ namespace iroha {
         processApplicableBlock(committed_block_variant);
       } else {
         auto missing_chain = downloadMissingChain(committed_block_variant);
-        notifier_.get_subscriber().on_next(missing_chain);
+        notifier_.get_subscriber().on_next(SynchronizationEvent{
```

Comment:
This call should be placed after blocks are applied to the storage. Because it is out of scope of this pull request and task, please create another task in JIRA and leave a TODO here.